frontend/src/components/TwoLevelDropdown.tsx

```diff
@@ -0,0 +1,295 @@
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+const css = stylesheet({
+  defaultFont: {
+    fontSize: '1rem',
+  },
+  dropdown: {
+    border: `1px solid ${color.lightGrey}`,
+    color: color.grey,
+    fontWeight: 400,
+    margin: 0,
+  },
+  dropdownMenu: {
+    display: 'block',
+    borderRadius: '5px',
+    border: `1px solid ${color.lightGrey}`,
+    margin: 0,
+    padding: '5px 0',
+    position: 'absolute',
+    top: '100%',
+    left: 0,
+    minWidth: '10rem',
+    cursor: 'pointer',
+    backgroundColor: color.background,
+  },
+  dropdownElement: {
+    padding: '1rem',
+    position: 'relative',
+
+    $nest: {
+      '&:hover': {
+        backgroundColor: `${color.lightGrey}`,
+      },
+    },
+  },
+  textContainer: {
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    maxWidth: '30rem',
+    overflow: 'hidden',
+    verticalAlign: 'top',
+  },
+  dropdownSubmenu: {
+    left: '100%',
+    // Offsets the dropdown menu vertical padding (5px) and border (1px).
+    top: '-6px',
+  },
+  inlineContainer: {
+    display: 'inline-block',
+  },
+  relativeContainer: {
+    position: 'relative',
+  },
+});
+
+const ChevronRightSmallIcon = () => <ChevronRightIcon className={classes(css.defaultFont)} />;
+
+export interface SelectedItem {
+  itemName: string;
+  subItemName: string;
+  subItemSecondaryName?: string;
+}
+
+interface DropdownSubItem {
+  name: string;
+  secondaryName?: string;
+}
+
+export interface DropdownItem {
+  name: string;
+  subItems: DropdownSubItem[];
+}
+
+interface DropdownButtonProps {
+  dropdownRef: Ref<HTMLDivElement>;
+  toggleDropdown: (e: React.MouseEvent) => void;
+  selectedItem: SelectedItem;
+  title: string;
+}
+
+function DropdownButton(props: DropdownButtonProps) {
+  const { dropdownRef, toggleDropdown, selectedItem, title } = props;
+
+  const dropdownTooltip: string =
+    selectedItem.itemName && selectedItem.subItemName
+      ? `${selectedItem.itemName} > ${selectedItem.subItemName}` +
+        (selectedItem.subItemSecondaryName ? ` > ${selectedItem.subItemSecondaryName}` : '')
+      : title;
+  return (
+    <div className={classes(css.inlineContainer)} ref={dropdownRef}>
+      <Button onClick={toggleDropdown} className={classes(css.dropdown, css.textContainer)}>
+        <Tooltip title={dropdownTooltip} enterDelay={300} placement='top-start'>
+          <span className={classes(css.textContainer)}>
+            {selectedItem.itemName && selectedItem.subItemName ? (
+              <>
+                {selectedItem.itemName}
+                <ChevronRightSmallIcon />
+                {selectedItem.subItemName}
+                {selectedItem.subItemSecondaryName && (
+                  <>
+                    <ChevronRightSmallIcon />
+                    {selectedItem.subItemSecondaryName}
+                  </>
+                )}
+              </>
```

Comment:
In the case of `subItem` doesn't have any `subItemSecondaryName` (aka Artifacts name), do we still need to render this `subItem`?

Reply:
I planned to still render `subItem` to try to generalize this component - if somebody else wants to use a two-level dropdown but does not want to use that `subItemSecondaryName` feature, then they can simply leave that `undefined` and only define the `subItemName`. I'm happy to not render the `subItem` in that case as you suggest though, if you think we might as well specialize this component further.

When creating the `DropdownItem[]` object from the `RunArtifacts[]` object, I plan to only pass in the `subItem` if I successfully find the execution name and artifact name. I have a draft of this implementation here: https://github.com/kubeflow/pipelines/pull/7943.